tests/UnitTests/Controllers/ProductControllerTests.cs

```diff
@@ -0,0 +1,197 @@
+
+            var actualResult = createdResult.Value as ProductOutbound;
+            Assert.IsInstanceOf<ProductOutbound>(actualResult);
+            Assert.AreEqual(expectedProductOutbound, actualResult);
+            _productServiceMock.Verify(x => x.AddItem(It.IsAny<ProductInbound>(), It.IsAny<CancellationToken>()), Times.Once);
+        }
+
+        [Test]
+        public async Task AddProduct_Returns_CorrectProductValues()
+        {
+            // Arrange
+            var productInbound = new ProductInbound
+            {
+                Name = $"The Three Musketeers {DateTime.UtcNow.Ticks}",
+                Description = $"You have likely heard of The Three Musketeers! {DateTime.UtcNow.Ticks}",
+                Author = $"Alexandre Dumas {DateTime.UtcNow.Ticks}",
+                Price = 12.56f,
+                ImageUrl = $"ftp://book.shop/{DateTime.UtcNow.Ticks}/image.jpg"
+            };
+
+            var expectedOutbound = new ProductOutbound
+            {
+                Id = Guid.NewGuid(),
+                Name = productInbound.Name,
+                Description = productInbound.Description,
+                Author = productInbound.Author,
+                Price = productInbound.Price,
+                ImageUrl = productInbound.ImageUrl,
+                BookingId = null
+            };
+
+            _productServiceMock.Setup(x => x.AddItem(productInbound, It.IsAny<CancellationToken>()))
+                               .ReturnsAsync(expectedOutbound);
+
+            // Act
+            var result = await _productController.AddProduct(productInbound);
+            var actualResult = ((CreatedAtActionResult)result).Value as ProductOutbound;
+
+            // Assert
+            Assert.IsNotNull(actualResult);
+            Assert.AreEqual(expectedOutbound.Id, actualResult.Id);
+            Assert.AreEqual(expectedOutbound.Name, actualResult.Name);
+            Assert.AreEqual(expectedOutbound.Description, actualResult.Description);
+            Assert.AreEqual(expectedOutbound.Author, actualResult.Author);
+            Assert.AreEqual(expectedOutbound.Price, actualResult.Price);
+            Assert.AreEqual(expectedOutbound.ImageUrl, actualResult.ImageUrl);
+            Assert.AreEqual(expectedOutbound.BookingId, actualResult.BookingId);
+        }
+
+        [Test]
+        [TestCase(0, "NotFound by id: 'guid'")]
+        [TestCase(1, "Product with id 'guid' was deleted")]
+        public async Task DeleteProduct_Returns_SimpleResult(int result, string expectedMessage)
+        {
+            // Arrange
+            Guid id = Guid.NewGuid();
+            _productServiceMock.Setup(x => x.RemoveItemById(id))
+                .ReturnsAsync(result);
+            expectedMessage = expectedMessage.Replace("guid", id.ToString());
+
+            // Act
+            var objectResult = await _productController.DeleteProduct(id);
+
+            // Assert
+            var responseBody = (SimpleResult)((ObjectResult)objectResult).Value;
```

Comment:
You could first check the result and responseBody on null. I see github actions are generating warnings because it's not done.